index.js

```diff
@@ -84,26 +85,35 @@ cli.on('respawn', function(flags, child) {
 });
 
 function run() {
-  cli.launch({
+  var envOpts = {
     cwd: opts.cwd,
     configPath: opts.gulpfile,
     require: opts.require,
     completion: opts.completion,
-  }, handleArguments);
+  };
+
+  cli.launch(envOpts, function(env) {
+    var config;
+    try {
+      config = loadConfigFiles(env.configFiles['.gulp'], ['home', 'cwd']);
+    } catch (e) {
+      log.error(chalk.red(e.message));
+      exit(1);
+    }
+
+    mergeToCliFlags(opts, config, cliOptions);
+    mergeToEnvFlags(env, config, envOpts);
```

Comment:
> Why do we pass envOpts in here?

To exclude `env` property names related to command-line arguments.

> This should be non-mutating (return a new object).

I got it.